### PR TITLE
MAINT: stats.boxcox_normmax: make default ymax value appear as BIG_FLOAT

### PR DESCRIPTION
**Pull request by mdhaber**
Re: https://github.com/scipy/scipy/pull/19631/files#r1414500609. The intent is to make the signature look like `ymax=BIG_FLOAT` (because `ymax=None` suggests that there is no finite `ymax`; i.e. `ymax=np,inf`).

### CODE REVIEW

File: scipy/stats/_morestats.py [hunk 1]
```diff
@@ -1129,12 +1129,21 @@ def boxcox(x, lmbda=None, alpha=None, optimizer=None):
 
 def _boxcox_inv_lmbda(x, y):
     # compute lmbda given x and y for Box-Cox transformation
-    num = special.lambertw(-(x ** (-1 / y)) * np.log(x) / y, k=1)
+    num = special.lambertw(-(x ** (-1 / y)) * np.log(x) / y, k=-1)
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:1132]: I'm going to put this in a separate PR. It looks like whether the `-1` or `0` branch should be chosen depends on `x`. This is a bug fix, so we need to get it in 112.

**Comment by xuefeng-xu** [comment]
Thanks for yours help, this LGTM except for some nits.

File: scipy/stats/_morestats.py [hunk 2]
```diff
@@ -1242,7 +1251,12 @@ def boxcox_normmax(
     >>> stats.boxcox_normmax(x, optimizer=optimizer)
     6.000...
     """
-    if ymax is not None and ymax <= 0:
+    x = np.asarray(x)
+    end_msg = "exceed specified `ymax`."
+    if isinstance(ymax, BigFloat):
+        ymax = ymax(x)
+        end_msg = f"overflow."
```
Comment (xuefeng-xu) [inline scipy/stats/_morestats.py:1258]: Do you think we need to add `dtype` to make it more informative?

File: scipy/stats/_morestats.py [hunk 3]
```diff
@@ -1321,19 +1335,10 @@ def _all(x):
         message = ("The `optimizer` argument of `boxcox_normmax` must return "
                    "an object containing the optimal `lmbda` in attribute `x`.")
         raise ValueError(message)
-    elif ymax is None or not np.isinf(ymax):  # adjust the final lambda
+    elif not np.isinf(ymax):  # adjust the final lambda
         x = np.asarray(x)
```
Comment (xuefeng-xu) [inline scipy/stats/_morestats.py]:
```suggestion

```

**Review by xuefeng-xu — APPROVED** [review]

File: scipy/stats/_morestats.py (hunk 2 above)
Comment (mdhaber) [reply scipy/stats/_morestats.py:1258]: Nope, that's why I took it out. But you're welcome to add it back after merging if you think it's important.

File: scipy/stats/_morestats.py (hunk 3 above)
Comment (mdhaber) [reply scipy/stats/_morestats.py]: Right,  I moved this but forgot to delete it here.
